lms/djangoapps/support/tests/test_views.py

```diff
@@ -1189,3 +1189,184 @@ def test_fbe_disabled_response(self):
+        """
+        program_enrollment = ProgramEnrollmentFactory.create(
+            external_user_key=external_user_key,
+            program_uuid=self.program_uuid,
+            user=linked_user
+        )
+        course_enrollment = None
+        if linked_user:
+            course_enrollment = CourseEnrollmentFactory.create(
+                course_id=self.course.id,
+                user=linked_user,
+                mode=CourseMode.MASTERS,
+                is_active=True
+            )
+        program_course_enrollment = ProgramCourseEnrollmentFactory.create(
+            program_enrollment=program_enrollment,
+            course_key=self.course.id,
+            course_enrollment=course_enrollment,
+            status='active'
+        )
+        return program_enrollment, program_course_enrollment
+
+    def test_invalid_uuid(self):
+        """
+        Tests if enrollment linkages are refused for an invalid uuid
+        """
+        response = self.client.post(self._url, data={
+            'program_uuid': 'notauuid',
+            'username_pair_text': self.username_pair_text,
+        })
+        msg = "Supplied program UUID 'notauuid' is not a valid UUID."
+        data = json.loads(response.content.decode('utf-8'))
+        assert data['errors'] == [msg]
+
+    @ddt.data(
+        ('program_uuid', ''),
+        ('', 'username_pair_text'),
+        ('', '')
+    )
+    @ddt.unpack
+    def test_missing_parameter(self, program_uuid, username_pair_text):
+        """
+        Tests if enrollment linkages are refused for missing parameters
+        """
+        error = (
+            "You must provide both a program uuid "
+            "and a series of lines with the format "
+            "'external_user_key,lms_username'."
+        )
+        response = self.client.post(self._url, data={
+            'program_uuid': program_uuid,
+            'username_pair_text': username_pair_text
+        })
+        response_data = json.loads(response.content.decode('utf-8'))
+        assert response_data['errors'] == [error]
+
+    @ddt.data(
+        '0001,learner-01\n0002,learner-02',                                 # normal
+        '0001,learner-01,apple,orange\n0002,learner-02,purple',             # extra fields
+        '\t0001        ,    \t  learner-01    \n   0002 , learner-02    ',  # whitespace
+    )
+    @patch('lms.djangoapps.support.views.utils.link_program_enrollments')
+    def test_username_pair_text(self, username_pair_text, mocked_link):
+        """
+        Tests if enrollment linkages are created for different types of
+        username_pair_text format
+        """
+        self.client.post(self._url, data={
+            'program_uuid': self.program_uuid,
+            'username_pair_text': username_pair_text,
+        })
+        response = self.client.post(self._url, data={
+            'program_uuid': self.program_uuid,
+            'username_pair_text': username_pair_text,
+        })
```

Comment:
2 api calls are being made here, which is resulting in test failure. Also, you should reset mock at the end to test to avoid any bad data between tests.